src/checker/mod.rs

```diff
@@ -102,17 +104,36 @@ impl Checker for Checkers {
     where
         'a: 's,
     {
-        let mut collective = Vec::<Suggestion<'s>>::with_capacity(chunks.len());
+        let mut collective = HashSet::<Suggestion<'s>>::new();
         if let Some(ref hunspell) = self.hunspell {
             collective.extend(hunspell.check(origin, chunks)?);
         }
         if let Some(ref nlprule) = self.nlprules {
             collective.extend(nlprule.check(origin, chunks)?);
         }
 
-        collective.sort();
+        let mut suggestions: Vec<Suggestion<'s>> = Vec::from_iter(collective);
+        suggestions.sort();
+        if suggestions.len() == 0 {
+            return Ok(suggestions);
+        }
+
+        // Iterate through suggestions and identify overlapping ones.
+        let mut omitted: Vec<CheckableChunk> = Vec::<CheckableChunk>::new();
+        let mut i: usize = 0;
+        while i < suggestions.len() - 1 as usize {
+            let cur = suggestions[i].clone();
+            i += 1;
+
+            while cur.is_overlapped(&suggestions[i]) {
+                let omit = suggestions.remove(i);
+                omitted.push(omit.chunk.clone());
+            }
+        }
```

Comment:
I'd recommend to use `suggestions.iter().filter_map(..)` in conjuction with a collection of overlapping ones.